# Curate Week 5 Assignment

#### Question 1: What is the northernmost airport in the United States?

In [1]:
import pandas as pd
a_d = pd.read_csv("airports.csv")

Check what the file looks like:

In [2]:
a_d.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

Since latitude represents the north/south orientation of a point, sorting for the highest latitude within the US should give us the northernmost airport.

In [3]:
a_d.sort_values('lat', ascending=False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

Need to remove the tzone with the null value and confirm the new record order.

In [4]:
a_d[a_d.tzone.notnull()].sort_values('lat', ascending=False).head()

faa                                        name        lat         lon  \
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
1363  UUK                        Ugnu-Kuparuk Airport  70.330833 -149.597500   

      alt  tz dst              tzone  
230    44  -9   A  America/Anchorage  
110    41  -9   A  America/Anchorage  
708    35  -9   A  America/Anchorage  
152    96  -9   A  America/Anchorage  
1363   67  -9   A  America/Anchorage

isolate the record with the highest lat

In [5]:
a_d[(a_d.lat == (a_d[a_d.tzone.notnull()].lat.max()))]

faa                        name        lat         lon  alt  tz dst  \
230  BRW  Wiley Post Will Rogers Mem  71.285446 -156.766003   44  -9   A   

                 tzone  
230  America/Anchorage

#### <font color= green> __Wiley Post Will Rogers Memorial Airport is the northernmost US Airport.__ </font>

The only third-party source I found that confims this is an obscure reference in this site:

http://travel.fyicenter.com/Airport/Top_30_Closest_Airports_to_the_North_Pole.php

#### Question 2: What is the easternmost airport in the United States? 

Sorting for the highest longitude should identify the easternmost airport. 

In [6]:
a_d.sort_values('lon', ascending=False).head()

faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York

Anchorage Alakska appears to be the answer, however, this is apparently debatable because it's actually beyond the prime meridian. So we'll focus on the contiguous US, which means we have to find lon with negative values:

In [7]:
a_d[(a_d.lon == (a_d[a_d.tzone.notnull() & (a_d.lon <= 0)].lon.max()))]

faa                        name        lat        lon  alt  tz dst  \
444  EPM  Eastport Municipal Airport  44.910111 -67.012694   45  -5   A   

                tzone  
444  America/New_York

#### <font color= green> __The eastern most airport in the US is Eastport Municipal Airport__ </font> 

https://www.thoughtco.com/states-farthest-north-south-east-west-4071599

#### Question 3: On February 12th, 2013, which New York area airport had the windiest weather? 

Need to grab the weather file and spot check the file contents:

In [8]:
w = pd.read_csv("weather.csv")

In [9]:
w.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z

Filtering for Feb 12, 2013:

In [10]:
w[(w.year == 2013) & (w.month == 2) & (w.day == 12)].head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006    EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007    EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008    EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1009    EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
1010    EWR  2013      2   12     4  39.02  26.96  64.29     280.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1006     6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1007     9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1008    20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
1009  1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
1010    12.65858        NaN     0.0    1008.8   10.0  2013-02-12T09:00:00Z

Add grouping to find the windiest weather by airport:

In [11]:
w[(w.year == 2013) & (w.month == 2) & (w.day == 12)].groupby('origin').wind_speed.max()

origin
EWR    1048.36058
JFK      20.71404
LGA      23.01560
Name: wind_speed, dtype: float64

Since hurricane force winds are less than 200 mph, the 1048 wind speed is probably just an error:

In [12]:
w[(w.year == 2013) & (w.month == 2) & (w.day == 12) & (w.wind_speed <= 200)].groupby('origin').wind_speed.max()

origin
EWR    21.86482
JFK    20.71404
LGA    23.01560
Name: wind_speed, dtype: float64

#### <font color= green> __LGA had the windiest weather on Feb 12, 2013 at 23 miles per hour.__ </font>